README.md

```diff
@@ -3,39 +3,77 @@
 A [Buildkite plugin](https://buildkite.com/docs/agent/v3/plugins) to allow pipeline steps to easily install
 private packages from an [npm](https://www.npmjs.com) repository.
 
-Note this plugin should work equally well despite your personal preferences for either `yarn` or `npm`.
+Note this plugin should work equally well despite any personal preferences for either `yarn` or `npm`.
 
 ## Example
 
-The following pipeline will run `yarn install` (and presumably some private packages).
+To read the value from an environment variable named `MY_TOKEN` when the plugin executes, use the `env:` prefix.
 
 ```yml
 steps:
   - command: yarn install
     plugins:
-      seek-oss/private-npm#v1.0.1:
-        token: ${NPM_TOKEN}
+      seek-oss/private-npm#v1.1.1:
+        token: "env:MY_TOKEN"
 ```
 
+To read the value from a file named `my_token_file`, use the `file:` prefix.
+
+```yml
+steps:
+  - command: yarn install
+    plugins:
+      seek-oss/private-npm#v1.1.1:
+        token: "file:my_token_file"
+```
+
+Alternatively you can read the token directly from any value exposed toxs your `pipeline.yml` file.  However, this 
+approach is discoraged in favour of using with the `env:` or `file:` prefix.  This functionality remains in the interest
+ of backwards compatibility.
+
+```yml
+steps:
+  - command: yarn install
+    plugins:
+      seek-oss/private-npm#v1.1.1:
+        token: ${MY_TOKEN}
+```
+
+
 You can also specify a custom npm registry if you are using your own mirror.
 
 ```yml
 steps:
   - command: yarn install
     plugins:
-      seek-oss/private-npm#v1.0.1:
-        token: ${NPM_TOKEN}
+      seek-oss/private-npm#v1.1.1:
+        token: ${MY_TOKEN}
```

Comment:
Maybe update this to the preferred approach?

Reply:
Not sure I follow you here, I think this is the variant we wish to deprecate in favour of `env` or `file`

Reply:
Oh right I'm with you now.  Good point I'll fix that.